ccl/backupccl: TestBackupRestoreChecksum failed

ccl/backupccl.TestBackupRestoreChecksum [failed](https://mesolite.cluster.engflow.com/invocations/default/d4de51e8-02d1-4af2-bc1f-43fb60001108?testReportRun=3&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on release-24.3 @ [a7392b107c050460829818af7ecb0cb94af6263f](https://github.com/cockroachdb/cockroach/commits/a7392b107c050460829818af7ecb0cb94af6263f):


```
=== RUN   TestBackupRestoreChecksum
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestBackupRestoreChecksum3812379094
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:152: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    backup_test.go:4135: expected query 'RESTORE data.* FROM $1' error 'checksum mismatch', got: pq: aborted in DistSender: result is ambiguous: context canceled
    testutils.go:302: no Invalid Descriptors
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestBackupRestoreChecksum3812379094
--- FAIL: TestBackupRestoreChecksum (14.56s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupRestoreChecksum.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-58134

--- a/pkg/backup/backup_test.go
+++ b/pkg/backup/backup_test.go
@@ -3968,6 +3968,8 @@ func TestBackupRestoreChecksum(t *testing.T) {
 
 	// Test Server too slow under deadlock.
 	skip.UnderDeadlock(t)
+	// Flaky under race.
+	skip.UnderRace(t)
 
 	const numAccounts = 1000
 	_, sqlDB, dir, cleanupFn := backupRestoreTestSetup(t, singleNode, numAccounts, InitManualReplication)

